### DISCUSSION TITLE
Jackson deserialization ignoring JsonProperty annotations

### DISCUSSION THREAD

**Discussion by rsenden** [micronaut-projects/micronaut-core#6470] [Q&A] [answered]
The following class is not properly deserialized when running as a native image; the properties are not properly being set due to the JsonProperty annotations on the setters being ignored:

```java
@Data @Introspected() @JsonIgnoreProperties(ignoreUnknown = true)
public final class TokenResponse {
	private String accessToken;
	private long expiresAt;
	public String getAccessToken() {
		return accessToken;
	}
	@JsonProperty("access_token")
	public void setAccessToken(String accessToken) {
		this.accessToken = accessToken;
	}
	@JsonProperty("expires_in")
	public void setExpiresIn(long expiresIn) {
		this.expiresAt = new Date().getTime()+((expiresIn-5)*1000);
	}
	public long getExpiresAt() {
		return expiresAt;
	}
	@JsonIgnore public boolean isExpired() {
		return new Date().getTime() > expiresAt;
	}
}
```

I would expect `@Introspected` to make the method annotations available to Jackson, but apparently this assumption is wrong. Looking at the generated introspection classes, they only list information about the class-level annotations and not about the method annotations.

I can easily fix this issue by adding the `@ReflectiveAccess` annotation to this class, or by renaming the methods to `setAccess_token()`/`setExpires_in()` and removing the `@JsonProperty` annotations altogether. 

The question is why I need to do this though? Is it expected behavior for Jackson not seeing these method annotations if the class has been annotated with `@Introspected`, or can this be considered a bug in Micronaut's `@Introspected` implementation?

**Comment by yawkat** [comment, marked as answer]
The introspection-based serialization used in native image (when reflective access is not available) mimics jackson behavior, but does not match it completely, so this sort of thing pops up regularly. Please report this as a bug and I will implement this.

**Reply by rsenden** [reply]
Thanks for the reply, reported as a bug: https://github.com/micronaut-projects/micronaut-core/issues/6472
